tests/actions/detectReactComponent.test.ts

```diff
@@ -0,0 +1,79 @@
+import {detectReactComponent} from '../../.github/actions/javascript/authorChecklist/categories/newComponentCategory';
+
+describe('detectReactComponent test', () => {
+    it('should return undefined if no code is provided', () => {
+        const result = detectReactComponent('', 'filename.js');
+
+        expect(result).toBeUndefined();
+    });
+
+    it('should detect if code is React component', () => {
+        const code = `
+            function Component() {
+                return <div>Hello World</div>;
+            }
+        `;
+        const result = detectReactComponent(code, 'filename.js');
+
+        expect(result).toBe(true);
+    });
+
+    it('should detect if code is a Arrow function component', () => {
+        const code = `
+            const Component = () => {
+                return <div>Hello World</div>;
+            }
+        `;
+        const result = detectReactComponent(code, 'filename.js');
+
+        expect(result).toBe(true);
+    });
+
+    it('should detect if code is in typescript', () => {
+        const code = `
+            const Component = ({text}: {text: string}) => {
+                return <div>{text}</div>;
+            }
+        `;
+        const result = detectReactComponent(code, 'filename.js');
+
+        expect(result).toBe(true);
+    });
+
+    it('should detect if code is a Class component', () => {
+        const code = `
+            class SomeComponent extends Component {
+                render() {
+                    return <div>Hello World</div>;
+                }
+            }
+        `;
+        const result = detectReactComponent(code, 'filename.js');
+
+        expect(result).toBe(true);
+    });
+
+    it('should detect if code is a Class component with React namespace', () => {
+        const code = `
+            class Component extends React.Component {
```

Comment:
Can you also add some more tests for edge cases please?

- `class MyComponent extends PureComponent`
- `class MyComponent extends React.PureComponent`
- `export default memo(MyComponent)`
- `export default React.forwardRef(...)`
- `export default forwardRef(...)`
- `export default withOnyx(...)`

Reply:
I've added test cases for PureComponents. However I'm not sure if we should add other cases - those are not components definitions, just using HOCs on some components that have been defined with function/arrow function/class elsewhere.